Libs/MRML/Core/vtkCodedEntry.cxx

```diff
@@ -0,0 +1,156 @@
+vtkCodedEntry::vtkCodedEntry()
+: CodingSchemeDesignator(NULL)
+, CodeValue(NULL)
+, CodeMeaning(NULL)
+{
+}
+
+//----------------------------------------------------------------------------
+vtkCodedEntry::~vtkCodedEntry()
+{
+  this->Initialize();
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::Initialize()
+{
+  this->SetCodingSchemeDesignator(NULL);
+  this->SetCodeValue(NULL);
+  this->SetCodeMeaning(NULL);
+}
+
+;
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::SetSchemeValueMeaning(const std::string& scheme,
+  const std::string& value, const std::string& meaning)
+{
+  this->SetCodingSchemeDesignator(scheme.c_str());
+  this->SetCodeValue(value.c_str());
+  this->SetCodeMeaning(meaning.c_str());
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::PrintSelf(ostream& os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os,indent);
+  os << indent << "CodingSchemeDesignator: " << (this->CodingSchemeDesignator?this->CodingSchemeDesignator:"(none)") << "\n";
+  os << indent << "CodeValue: " << (this->CodeValue?this->CodeValue:"(none)") << "\n";
+  os << indent << "CodeMeaning: " << (this->CodeMeaning?this->CodeMeaning:"(none)") << "\n";
+}
+
+//----------------------------------------------------------------------------
+void vtkCodedEntry::Copy(vtkCodedEntry* aType)
+{
+  if (!aType)
+    {
+    return;
+    }
+  this->SetCodingSchemeDesignator(aType->GetCodingSchemeDesignator());
+  this->SetCodeValue(aType->GetCodeValue());
+  this->SetCodeMeaning(aType->GetCodeMeaning());
+}
+
+//----------------------------------------------------------------------------
+std::string vtkCodedEntry::GetAsPrintableString()
+{
+  std::string printable = std::string("(")
+    + (this->CodingSchemeDesignator ? this->CodingSchemeDesignator : "(none)") + ", "
+    + (this->CodeValue ? this->CodeValue : "(none)") + ", "
+    + (this->CodeMeaning ? this->CodeMeaning : "(none)") + ")";
+  return printable;
+}
+
+//----------------------------------------------------------------------------
+std::string vtkCodedEntry::GetAsString()
+{
+  std::string str;
+  if (this->CodingSchemeDesignator)
+    {
+    str += "CodingSchemeDesignator:";
+    str += this->CodingSchemeDesignator;
+    }
+  if (this->CodeValue)
+    {
+    if (!str.empty())
+      {
+      str += "|";
+      }
+    str += "CodeValue:";
+    str += this->CodeValue;
+    }
+  if (this->CodeMeaning)
+    {
+    if (!str.empty())
+      {
+      str += "|";
+      }
+    str += "CodeMeaning:";
+    str += this->CodeMeaning;
+    }
+  return str;
+}
+
+//----------------------------------------------------------------------------
+bool vtkCodedEntry::SetFromString(const std::string& content)
+{
+  this->Initialize();
+  bool success = true;
+  std::stringstream attributes(content);
+  std::string attribute;
```

Comment:
Currently, if more fields are defined then a warning is logged for each unknown fields and the method returns with false. Should we change this behavior?